scripts/validate.py

```diff
@@ -0,0 +1,209 @@
+
+    indexd_records = get_indexd_records()
+    aws_copied_objects, _ = build_object_dataset(PROJECT_ACL, logger)
+    gs_copied_objects = utils.build_object_dataset_gs(PROJECT_ACL)
+
+    if global_config.get("save_copied_objects"):
+        with open("./indexd_records.json", "w") as outfile:
+            json.dump(indexd_records, outfile)
+        with open("./aws_copied_objects.json", "w") as outfile:
+            json.dump(aws_copied_objects, outfile)
+        with open("./gs_copied_objects.json", "w") as outfile:
+            json.dump(gs_copied_objects, outfile)
+
+        s3 = boto3.client("s3")
+        try:
+            s3.upload_file(
+                "indexd_records.json",
+                global_config.get("log_bucket"),
+                "indexd_records.json",
+            )
+            s3.upload_file(
+                "aws_copied_objects.json",
+                global_config.get("log_bucket"),
+                "aws_copied_objects.json",
+            )
+            s3.upload_file(
+                "gs_copied_objects.json",
+                global_config.get("log_bucket"),
+                "gs_copied_objects.json",
+            )
+        except Exception as e:
+            logger.error(e)
+
+    pass_valiation = True
+    for idx, manifest_file in enumerate(manifest_files):
+        total_aws_copy_failures = 0
+        total_gs_copy_failures = 0
+        total_aws_index_failures = 0
+        total_gs_index_failures = 0
+        manifest_file = manifest_file.strip()
+        files = utils.get_fileinfo_list_from_s3_manifest(manifest_file)
+        for fi in files:
+            del fi["url"]
+            fi["aws_url"], fi["gs_url"], fi["indexd_url"] = None, None, None
+
+            fi["indexd_url"] = indexd_records.get(fi.get("id"))
+            if not fi["indexd_url"]:
+                total_aws_index_failures += 1
+                total_gs_index_failures += 1
+                logger.error("There is no indexd record for {}".format(fi["id"]))
+
+            # validate aws
+            aws_bucket = utils.get_aws_bucket_name(fi, PROJECT_ACL)
+            object_path = "{}/{}/{}".format(aws_bucket, fi["id"], fi["file_name"])
+            if object_path not in aws_copied_objects and fi["size"] != 0:
+                total_aws_copy_failures += 1
+                logger.error("{} is not copied yet to aws buckets".format(object_path))
+            elif fi["size"] != 0:
+                fi["aws_url"] = "s3://" + object_path
+                if fi["aws_url"] not in fi["indexd_url"]:
+                    total_aws_index_failures += 1
+                    logger.error("indexd does not have aws url of {}".format(fi["id"]))
+
+            # validate google
+            gs_bucket = utils.get_google_bucket_name(fi, PROJECT_ACL)
+            if fi["id"] in ignored_dict:
+                object_path = "{}/{}".format(
+                    gs_bucket, utils.get_structured_object_key(fi["id"], ignored_dict)
+                )
```

Comment:
maybe add in a comment what you told me? `if a file is in ignored_dict, we don’t copy but we still need to check if the file exists in the bucket and is indexed`